File: app/src/main/java/com/example/gigmarket/data/model/Booking.kt
package com.example.gigmarket.data.model

data class Booking(
    val _id: String,
    val userId: UserSummary?,
    val providerId: ProviderSummary?,
    val status: String, // Pending, Accepted, Completed, Cancelled
    val bookingDate: String,
    val address: String,
    val notes: String?
)

data class UserSummary(
    val _id: String,
    val name: String,
    val phone: String
)

data class ProviderSummary(
    val _id: String,
    val name: String,
    val category: String,
    val hourlyRate: Double
)

data class BookingRequest(
    val providerId: String,
    val address: String,
    val notes: String?
)

data class BookingResponse(
    val success: Boolean,
    val booking: Booking
)

data class BookingsListResponse(
    val success: Boolean,
    val bookings: List<Booking>
)

File: server/server/models/Booking.js
const mongoose = require("mongoose");

const bookingSchema = new mongoose.Schema({
    userId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
    providerId: { type: mongoose.Schema.Types.ObjectId, ref: "Provider", required: true },
    status: {
        type: String,
        enum: ["Pending", "Accepted", "Completed", "Cancelled"],
        default: "Pending"
    },
    bookingDate: { type: Date, default: Date.now },
    address: String,
    notes: String,
}, { timestamps: true });

module.exports = mongoose.model("Booking", bookingSchema);
